File: examples/numerical_propagation/force_model_gravity_pointmass.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Configuring point mass gravity for the force model.
Simplest gravity configuration with no perturbations.
"""

import brahe as bh

# Point mass gravity is configured using GravityConfiguration.point_mass()
# This uses only central body gravity (mu/r^2) - no spherical harmonics

# Create point mass gravity configuration
gravity = bh.GravityConfiguration.point_mass()

# Use two_body() preset which includes point mass gravity
force_config = bh.ForceModelConfig.two_body()

print("Point Mass Gravity Configuration:")
print(f"  Gravity is point mass: {gravity.is_point_mass()}")
print(f"  Gravity is spherical harmonic: {gravity.is_spherical_harmonic()}")

File: examples/numerical_propagation/force_model_gravity_pointmass.rs
//! Configuring point mass gravity for the force model.
//! Simplest gravity configuration with no perturbations.

use brahe as bh;

fn main() {
    // Point mass gravity configuration
    // Uses only central body gravity (mu/r^2)
    // No spherical harmonics, J2, or higher-order terms
    let force_config = bh::ForceModelConfiguration {
        gravity: bh::GravityConfiguration::PointMass,
        drag: None,
        srp: None,
        third_body: None,
        relativity: false,
        mass: None,
    };

    // This is equivalent to using the two_body_gravity() preset
    let preset_config = bh::ForceModelConfiguration::two_body_gravity();
    println!("\nEquivalent preset: ForceModelConfiguration::two_body_gravity()");
    println!("  Preset requires params: {}", preset_config.requires_params());
}
